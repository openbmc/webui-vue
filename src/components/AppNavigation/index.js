import AppNavigation from "./AppNavigation";
export default AppNavigation;
