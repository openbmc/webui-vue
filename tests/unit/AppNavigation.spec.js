import { mount, createWrapper } from '@vue/test-utils';
import AppNavigation from '@/components/AppNavigation';
import Vue from 'vue';
import VueRouter from 'vue-router';
import { BootstrapVue } from 'bootstrap-vue';

describe('AppNavigation.vue', () => {
  let wrapper;
  Vue.use(BootstrapVue);
  Vue.use(VueRouter);
  const router = new VueRouter();

  wrapper = mount(AppNavigation, {
    router,
    mocks: {
      $t: (key) => key,
    },
  });

  it('should exist', async () => {
    expect(wrapper.exists()).toBe(true);
  });

  it('should render correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render with nav-container open', () => {
    wrapper.vm.isNavigationOpen = true;
    expect(wrapper.element).toMatchSnapshot();
  });

  it('Nav Overlay click should emit change-is-navigation-open event', async () => {
    const rootWrapper = createWrapper(wrapper.vm.$root);
    const navOverlay = wrapper.find('#nav-overlay');
    navOverlay.trigger('click');
    await wrapper.vm.$nextTick();
    expect(rootWrapper.emitted('change-is-navigation-open')).toBeTruthy();
  });

  it('toggle-navigation event should toggle isNavigation data prop value', async () => {
    const rootWrapper = createWrapper(wrapper.vm.$root);
    wrapper.vm.isNavigationOpen = false;
    rootWrapper.vm.$emit('toggle-navigation');
    expect(wrapper.vm.isNavigationOpen).toBe(true);
    rootWrapper.vm.$emit('toggle-navigation');
    expect(wrapper.vm.isNavigationOpen).toBe(false);
  });
});
