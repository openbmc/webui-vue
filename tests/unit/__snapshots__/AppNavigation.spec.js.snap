// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppNavigation.vue should render correctly 1`] = `
<div>
  <div
    class="nav-container"
  >
    <nav
      aria-label="appNavigation.primaryNavigation"
    >
      <ul
        class="nav mb-4 flex-column"
      >
        <li
          class="nav-item"
          data-test-id="nav-item-overview"
        >
          <a
            class="nav-link"
            href="/"
            target="_self"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              focusable="false"
              height="16"
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 32 32"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4,2H2V28a2,2,0,0,0,2,2H30V28H4Z"
              />
              <path
                d="M30,9H23v2h3.59L19,18.59l-4.29-4.3a1,1,0,0,0-1.42,0L6,21.59,7.41,23,14,16.41l4.29,4.3a1,1,0,0,0,1.42,0L28,12.41V16h2Z"
              />
            </svg>
            
            appNavigation.overview
          
          </a>
        </li>
        <li
          class="nav-item"
        >
          <button
            aria-controls="health"
            aria-expanded="false"
            class="btn btn-link collapsed"
            data-test-id="nav-button-health"
            type="button"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              focusable="false"
              height="16"
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 32 32"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M23 27.18L20.41 24.59 19 26 23 30 30 23 28.59 21.59 23 27.18z"
              />
              <circle
                cx="11"
                cy="8"
                r="1"
              />
              <circle
                cx="11"
                cy="16"
                r="1"
              />
              <circle
                cx="11"
                cy="24"
                r="1"
              />
              <path
                d="M24,3H8A2,2,0,0,0,6,5V27a2,2,0,0,0,2,2h8V27H8V21H26V5A2,2,0,0,0,24,3Zm0,16H8V13H24Zm0-8H8V5H24Z"
              />
            </svg>
            
              appNavigation.health
              
            <svg
              aria-hidden="true"
              class="icon-expand"
              fill="currentColor"
              focusable="false"
              height="16"
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8 5L13 10 12.3 10.7 8 6.4 3.7 10.7 3 10z"
              />
            </svg>
          </button>
           
          <transition-stub
            class="nav-item__nav"
            css="true"
            enteractiveclass="collapsing"
            enterclass=""
            entertoclass="collapse show"
            leaveactiveclass="collapsing"
            leaveclass="collapse show"
            leavetoclass="collapse"
          >
            <ul
              class="collapse"
              id="health"
              style="display: none;"
            >
              <li
                class="nav-item"
                data-test-id="nav-item-event-logs"
              >
                <a
                  class="nav-link"
                  href="/health/event-logs"
                  target="_self"
                >
                  
                appNavigation.eventLogs
              
                </a>
              </li>
              <li
                class="nav-item"
                data-test-id="nav-item-hardware-status"
              >
                <a
                  class="nav-link"
                  href="/health/hardware-status"
                  target="_self"
                >
                  
                appNavigation.hardwareStatus
              
                </a>
              </li>
              <li
                class="nav-item"
                data-test-id="nav-item-sensors"
              >
                <a
                  class="nav-link"
                  href="/health/sensors"
                  target="_self"
                >
                  
                appNavigation.sensors
              
                </a>
              </li>
            </ul>
          </transition-stub>
        </li>
        <li
          class="nav-item"
        >
          <button
            aria-controls="control"
            aria-expanded="false"
            class="btn btn-link collapsed"
            data-test-id="nav-button-control"
            type="button"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              focusable="false"
              height="16"
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 32 32"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M30 8h-4.1c-.5-2.3-2.5-4-4.9-4s-4.4 1.7-4.9 4H2v2h14.1c.5 2.3 2.5 4 4.9 4s4.4-1.7 4.9-4H30V8zM21 12c-1.7 0-3-1.3-3-3s1.3-3 3-3 3 1.3 3 3S22.7 12 21 12zM2 24h4.1c.5 2.3 2.5 4 4.9 4s4.4-1.7 4.9-4H30v-2H15.9c-.5-2.3-2.5-4-4.9-4s-4.4 1.7-4.9 4H2V24zM11 20c1.7 0 3 1.3 3 3s-1.3 3-3 3-3-1.3-3-3S9.3 20 11 20z"
              />
            </svg>
            
              appNavigation.control
              
            <svg
              aria-hidden="true"
              class="icon-expand"
              fill="currentColor"
              focusable="false"
              height="16"
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8 5L13 10 12.3 10.7 8 6.4 3.7 10.7 3 10z"
              />
            </svg>
          </button>
           
          <transition-stub
            class="nav-item__nav"
            css="true"
            enteractiveclass="collapsing"
            enterclass=""
            entertoclass="collapse show"
            leaveactiveclass="collapsing"
            leaveclass="collapse show"
            leavetoclass="collapse"
          >
            <ul
              class="collapse"
              id="control"
              style="display: none;"
            >
              <li
                class="nav-item"
                data-test-id="nav-item-kvm"
              >
                <a
                  class="nav-link"
                  href="/control/kvm"
                  target="_self"
                >
                  
                appNavigation.kvm
              
                </a>
              </li>
              <li
                class="nav-item"
                data-test-id="nav-item-manage-power-usage"
              >
                <a
                  class="nav-link"
                  href="/control/manage-power-usage"
                  target="_self"
                >
                  
                appNavigation.managePowerUsage
              
                </a>
              </li>
              <li
                class="nav-item"
                data-test-id="nav-item-reboot-bmc"
              >
                <a
                  class="nav-link"
                  href="/control/reboot-bmc"
                  target="_self"
                >
                  
                appNavigation.rebootBmc
              
                </a>
              </li>
              <li
                class="nav-item"
                data-test-id="nav-item-serial-over-lan"
              >
                <a
                  class="nav-link"
                  href="/control/serial-over-lan"
                  target="_self"
                >
                  
                appNavigation.serialOverLan
              
                </a>
              </li>
              <li
                class="nav-item"
                data-test-id="nav-item-server-led"
              >
                <a
                  class="nav-link"
                  href="/control/server-led"
                  target="_self"
                >
                  
                appNavigation.serverLed
              
                </a>
              </li>
              <li
                class="nav-item"
                data-test-id="nav-item-server-power-operations"
              >
                <a
                  class="nav-link"
                  href="/control/server-power-operations"
                  target="_self"
                >
                  
                appNavigation.serverPowerOperations
              
                </a>
              </li>
              <li
                class="nav-item"
                data-test-id="nav-item-virtual-media"
              >
                <a
                  class="nav-link"
                  href="/control/virtual-media"
                  target="_self"
                >
                  
                appNavigation.virtualMedia
              
                </a>
              </li>
            </ul>
          </transition-stub>
        </li>
        <li
          class="nav-item"
        >
          <button
            aria-controls="configuration"
            aria-expanded="false"
            class="btn btn-link collapsed"
            data-test-id="nav-button-configuration"
            type="button"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              focusable="false"
              height="16"
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M13.5,8.4c0-0.1,0-0.3,0-0.4c0-0.1,0-0.3,0-0.4l1-0.8c0.4-0.3,0.4-0.9,0.2-1.3l-1.2-2C13.3,3.2,13,3,12.6,3	c-0.1,0-0.2,0-0.3,0.1l-1.2,0.4c-0.2-0.1-0.4-0.3-0.7-0.4l-0.3-1.3C10.1,1.3,9.7,1,9.2,1H6.8c-0.5,0-0.9,0.3-1,0.8L5.6,3.1	C5.3,3.2,5.1,3.3,4.9,3.4L3.7,3C3.6,3,3.5,3,3.4,3C3,3,2.7,3.2,2.5,3.5l-1.2,2C1.1,5.9,1.2,6.4,1.6,6.8l0.9,0.9c0,0.1,0,0.3,0,0.4	c0,0.1,0,0.3,0,0.4L1.6,9.2c-0.4,0.3-0.5,0.9-0.2,1.3l1.2,2C2.7,12.8,3,13,3.4,13c0.1,0,0.2,0,0.3-0.1l1.2-0.4	c0.2,0.1,0.4,0.3,0.7,0.4l0.3,1.3c0.1,0.5,0.5,0.8,1,0.8h2.4c0.5,0,0.9-0.3,1-0.8l0.3-1.3c0.2-0.1,0.4-0.2,0.7-0.4l1.2,0.4	c0.1,0,0.2,0.1,0.3,0.1c0.4,0,0.7-0.2,0.9-0.5l1.1-2c0.2-0.4,0.2-0.9-0.2-1.3L13.5,8.4z M12.6,12l-1.7-0.6c-0.4,0.3-0.9,0.6-1.4,0.8	L9.2,14H6.8l-0.4-1.8c-0.5-0.2-0.9-0.5-1.4-0.8L3.4,12l-1.2-2l1.4-1.2c-0.1-0.5-0.1-1.1,0-1.6L2.2,6l1.2-2l1.7,0.6	C5.5,4.2,6,4,6.5,3.8L6.8,2h2.4l0.4,1.8c0.5,0.2,0.9,0.5,1.4,0.8L12.6,4l1.2,2l-1.4,1.2c0.1,0.5,0.1,1.1,0,1.6l1.4,1.2L12.6,12z"
              />
              <path
                d="M8,11c-1.7,0-3-1.3-3-3s1.3-3,3-3s3,1.3,3,3C11,9.6,9.7,11,8,11C8,11,8,11,8,11z M8,6C6.9,6,6,6.8,6,7.9C6,7.9,6,8,6,8	c0,1.1,0.8,2,1.9,2c0,0,0.1,0,0.1,0c1.1,0,2-0.8,2-1.9c0,0,0-0.1,0-0.1C10,6.9,9.2,6,8,6C8.1,6,8,6,8,6z"
              />
            </svg>
            
              appNavigation.configuration
              
            <svg
              aria-hidden="true"
              class="icon-expand"
              fill="currentColor"
              focusable="false"
              height="16"
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8 5L13 10 12.3 10.7 8 6.4 3.7 10.7 3 10z"
              />
            </svg>
          </button>
           
          <transition-stub
            class="nav-item__nav"
            css="true"
            enteractiveclass="collapsing"
            enterclass=""
            entertoclass="collapse show"
            leaveactiveclass="collapsing"
            leaveclass="collapse show"
            leavetoclass="collapse"
          >
            <ul
              class="collapse"
              id="configuration"
              style="display: none;"
            >
              <li
                class="nav-item"
                data-test-id="nav-item-date-time-settings"
              >
                <a
                  class="nav-link"
                  href="/configuration/date-time-settings"
                  target="_self"
                >
                  
                appNavigation.dateTimeSettings
              
                </a>
              </li>
              <li
                class="nav-item"
                data-test-id="nav-item-firmware"
              >
                <a
                  class="nav-link"
                  href="/configuration/firmware"
                  target="_self"
                >
                  
                appNavigation.firmware
              
                </a>
              </li>
              <li
                class="nav-item"
                data-test-id="nav-item-network-settings"
              >
                <a
                  class="nav-link"
                  href="/configuration/network-settings"
                  target="_self"
                >
                  
                appNavigation.networkSettings
              
                </a>
              </li>
              <li
                class="nav-item"
                data-test-id="nav-item-snmp-settings"
              >
                <a
                  class="nav-link"
                  href="/snmp-settings"
                  target="_self"
                >
                  
                appNavigation.snmpSettings
              
                </a>
              </li>
            </ul>
          </transition-stub>
        </li>
        <li
          class="nav-item"
        >
          <button
            aria-controls="access-control"
            aria-expanded="false"
            class="btn btn-link collapsed"
            data-test-id="nav-button-access-control"
            type="button"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              focusable="false"
              height="16"
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 32 32"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M21,2a9,9,0,0,0-9,9,8.87,8.87,0,0,0,.39,2.61L2,24v6H8L18.39,19.61A9,9,0,0,0,30,11.74a8.77,8.77,0,0,0-1.65-6A9,9,0,0,0,21,2Zm0,16a7,7,0,0,1-2-.3l-1.15-.35L17,18.2l-3.18,3.18L12.41,20,11,21.41l1.38,1.38-1.59,1.59L9.41,23,8,24.41l1.38,1.38L7.17,28H4V24.83L13.8,15l.85-.85-.29-.95a7.14,7.14,0,0,1,3.4-8.44,7,7,0,0,1,10.24,6,6.69,6.69,0,0,1-1.09,4A7,7,0,0,1,21,18Z"
              />
              <circle
                cx="22"
                cy="10"
                r="2"
              />
            </svg>
            
              appNavigation.accessControl
              
            <svg
              aria-hidden="true"
              class="icon-expand"
              fill="currentColor"
              focusable="false"
              height="16"
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8 5L13 10 12.3 10.7 8 6.4 3.7 10.7 3 10z"
              />
            </svg>
          </button>
           
          <transition-stub
            class="nav-item__nav"
            css="true"
            enteractiveclass="collapsing"
            enterclass=""
            entertoclass="collapse show"
            leaveactiveclass="collapsing"
            leaveclass="collapse show"
            leavetoclass="collapse"
          >
            <ul
              class="collapse"
              id="access-control"
              style="display: none;"
            >
              <li
                class="nav-item"
                data-test-id="nav-item-ldap"
              >
                <a
                  class="nav-link"
                  href="/access-control/ldap"
                  target="_self"
                >
                  
                appNavigation.ldap
              
                </a>
              </li>
              <li
                class="nav-item"
                data-test-id="nav-item-local-user-management"
              >
                <a
                  class="nav-link"
                  href="/access-control/local-user-management"
                  target="_self"
                >
                  
                appNavigation.localUserManagement
              
                </a>
              </li>
              <li
                class="nav-item"
                data-test-id="nav-item-ssl-certificates"
              >
                <a
                  class="nav-link"
                  href="/access-control/ssl-certificates"
                  target="_self"
                >
                  
                appNavigation.sslCertificates
              
                </a>
              </li>
            </ul>
          </transition-stub>
        </li>
      </ul>
    </nav>
  </div>
   
  <transition-stub
    name="fade"
  >
    <!---->
  </transition-stub>
</div>
`;

exports[`AppNavigation.vue should render with nav-container open 1`] = `
<div>
  <div
    class="nav-container"
  >
    <nav
      aria-label="appNavigation.primaryNavigation"
    >
      <ul
        class="nav mb-4 flex-column"
      >
        <li
          class="nav-item"
          data-test-id="nav-item-overview"
        >
          <a
            class="nav-link"
            href="/"
            target="_self"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              focusable="false"
              height="16"
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 32 32"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4,2H2V28a2,2,0,0,0,2,2H30V28H4Z"
              />
              <path
                d="M30,9H23v2h3.59L19,18.59l-4.29-4.3a1,1,0,0,0-1.42,0L6,21.59,7.41,23,14,16.41l4.29,4.3a1,1,0,0,0,1.42,0L28,12.41V16h2Z"
              />
            </svg>
            
            appNavigation.overview
          
          </a>
        </li>
        <li
          class="nav-item"
        >
          <button
            aria-controls="health"
            aria-expanded="false"
            class="btn btn-link collapsed"
            data-test-id="nav-button-health"
            type="button"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              focusable="false"
              height="16"
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 32 32"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M23 27.18L20.41 24.59 19 26 23 30 30 23 28.59 21.59 23 27.18z"
              />
              <circle
                cx="11"
                cy="8"
                r="1"
              />
              <circle
                cx="11"
                cy="16"
                r="1"
              />
              <circle
                cx="11"
                cy="24"
                r="1"
              />
              <path
                d="M24,3H8A2,2,0,0,0,6,5V27a2,2,0,0,0,2,2h8V27H8V21H26V5A2,2,0,0,0,24,3Zm0,16H8V13H24Zm0-8H8V5H24Z"
              />
            </svg>
            
              appNavigation.health
              
            <svg
              aria-hidden="true"
              class="icon-expand"
              fill="currentColor"
              focusable="false"
              height="16"
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8 5L13 10 12.3 10.7 8 6.4 3.7 10.7 3 10z"
              />
            </svg>
          </button>
           
          <transition-stub
            class="nav-item__nav"
            css="true"
            enteractiveclass="collapsing"
            enterclass=""
            entertoclass="collapse show"
            leaveactiveclass="collapsing"
            leaveclass="collapse show"
            leavetoclass="collapse"
          >
            <ul
              class="collapse"
              id="health"
              style="display: none;"
            >
              <li
                class="nav-item"
                data-test-id="nav-item-event-logs"
              >
                <a
                  class="nav-link"
                  href="/health/event-logs"
                  target="_self"
                >
                  
                appNavigation.eventLogs
              
                </a>
              </li>
              <li
                class="nav-item"
                data-test-id="nav-item-hardware-status"
              >
                <a
                  class="nav-link"
                  href="/health/hardware-status"
                  target="_self"
                >
                  
                appNavigation.hardwareStatus
              
                </a>
              </li>
              <li
                class="nav-item"
                data-test-id="nav-item-sensors"
              >
                <a
                  class="nav-link"
                  href="/health/sensors"
                  target="_self"
                >
                  
                appNavigation.sensors
              
                </a>
              </li>
            </ul>
          </transition-stub>
        </li>
        <li
          class="nav-item"
        >
          <button
            aria-controls="control"
            aria-expanded="false"
            class="btn btn-link collapsed"
            data-test-id="nav-button-control"
            type="button"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              focusable="false"
              height="16"
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 32 32"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M30 8h-4.1c-.5-2.3-2.5-4-4.9-4s-4.4 1.7-4.9 4H2v2h14.1c.5 2.3 2.5 4 4.9 4s4.4-1.7 4.9-4H30V8zM21 12c-1.7 0-3-1.3-3-3s1.3-3 3-3 3 1.3 3 3S22.7 12 21 12zM2 24h4.1c.5 2.3 2.5 4 4.9 4s4.4-1.7 4.9-4H30v-2H15.9c-.5-2.3-2.5-4-4.9-4s-4.4 1.7-4.9 4H2V24zM11 20c1.7 0 3 1.3 3 3s-1.3 3-3 3-3-1.3-3-3S9.3 20 11 20z"
              />
            </svg>
            
              appNavigation.control
              
            <svg
              aria-hidden="true"
              class="icon-expand"
              fill="currentColor"
              focusable="false"
              height="16"
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8 5L13 10 12.3 10.7 8 6.4 3.7 10.7 3 10z"
              />
            </svg>
          </button>
           
          <transition-stub
            class="nav-item__nav"
            css="true"
            enteractiveclass="collapsing"
            enterclass=""
            entertoclass="collapse show"
            leaveactiveclass="collapsing"
            leaveclass="collapse show"
            leavetoclass="collapse"
          >
            <ul
              class="collapse"
              id="control"
              style="display: none;"
            >
              <li
                class="nav-item"
                data-test-id="nav-item-kvm"
              >
                <a
                  class="nav-link"
                  href="/control/kvm"
                  target="_self"
                >
                  
                appNavigation.kvm
              
                </a>
              </li>
              <li
                class="nav-item"
                data-test-id="nav-item-manage-power-usage"
              >
                <a
                  class="nav-link"
                  href="/control/manage-power-usage"
                  target="_self"
                >
                  
                appNavigation.managePowerUsage
              
                </a>
              </li>
              <li
                class="nav-item"
                data-test-id="nav-item-reboot-bmc"
              >
                <a
                  class="nav-link"
                  href="/control/reboot-bmc"
                  target="_self"
                >
                  
                appNavigation.rebootBmc
              
                </a>
              </li>
              <li
                class="nav-item"
                data-test-id="nav-item-serial-over-lan"
              >
                <a
                  class="nav-link"
                  href="/control/serial-over-lan"
                  target="_self"
                >
                  
                appNavigation.serialOverLan
              
                </a>
              </li>
              <li
                class="nav-item"
                data-test-id="nav-item-server-led"
              >
                <a
                  class="nav-link"
                  href="/control/server-led"
                  target="_self"
                >
                  
                appNavigation.serverLed
              
                </a>
              </li>
              <li
                class="nav-item"
                data-test-id="nav-item-server-power-operations"
              >
                <a
                  class="nav-link"
                  href="/control/server-power-operations"
                  target="_self"
                >
                  
                appNavigation.serverPowerOperations
              
                </a>
              </li>
              <li
                class="nav-item"
                data-test-id="nav-item-virtual-media"
              >
                <a
                  class="nav-link"
                  href="/control/virtual-media"
                  target="_self"
                >
                  
                appNavigation.virtualMedia
              
                </a>
              </li>
            </ul>
          </transition-stub>
        </li>
        <li
          class="nav-item"
        >
          <button
            aria-controls="configuration"
            aria-expanded="false"
            class="btn btn-link collapsed"
            data-test-id="nav-button-configuration"
            type="button"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              focusable="false"
              height="16"
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M13.5,8.4c0-0.1,0-0.3,0-0.4c0-0.1,0-0.3,0-0.4l1-0.8c0.4-0.3,0.4-0.9,0.2-1.3l-1.2-2C13.3,3.2,13,3,12.6,3	c-0.1,0-0.2,0-0.3,0.1l-1.2,0.4c-0.2-0.1-0.4-0.3-0.7-0.4l-0.3-1.3C10.1,1.3,9.7,1,9.2,1H6.8c-0.5,0-0.9,0.3-1,0.8L5.6,3.1	C5.3,3.2,5.1,3.3,4.9,3.4L3.7,3C3.6,3,3.5,3,3.4,3C3,3,2.7,3.2,2.5,3.5l-1.2,2C1.1,5.9,1.2,6.4,1.6,6.8l0.9,0.9c0,0.1,0,0.3,0,0.4	c0,0.1,0,0.3,0,0.4L1.6,9.2c-0.4,0.3-0.5,0.9-0.2,1.3l1.2,2C2.7,12.8,3,13,3.4,13c0.1,0,0.2,0,0.3-0.1l1.2-0.4	c0.2,0.1,0.4,0.3,0.7,0.4l0.3,1.3c0.1,0.5,0.5,0.8,1,0.8h2.4c0.5,0,0.9-0.3,1-0.8l0.3-1.3c0.2-0.1,0.4-0.2,0.7-0.4l1.2,0.4	c0.1,0,0.2,0.1,0.3,0.1c0.4,0,0.7-0.2,0.9-0.5l1.1-2c0.2-0.4,0.2-0.9-0.2-1.3L13.5,8.4z M12.6,12l-1.7-0.6c-0.4,0.3-0.9,0.6-1.4,0.8	L9.2,14H6.8l-0.4-1.8c-0.5-0.2-0.9-0.5-1.4-0.8L3.4,12l-1.2-2l1.4-1.2c-0.1-0.5-0.1-1.1,0-1.6L2.2,6l1.2-2l1.7,0.6	C5.5,4.2,6,4,6.5,3.8L6.8,2h2.4l0.4,1.8c0.5,0.2,0.9,0.5,1.4,0.8L12.6,4l1.2,2l-1.4,1.2c0.1,0.5,0.1,1.1,0,1.6l1.4,1.2L12.6,12z"
              />
              <path
                d="M8,11c-1.7,0-3-1.3-3-3s1.3-3,3-3s3,1.3,3,3C11,9.6,9.7,11,8,11C8,11,8,11,8,11z M8,6C6.9,6,6,6.8,6,7.9C6,7.9,6,8,6,8	c0,1.1,0.8,2,1.9,2c0,0,0.1,0,0.1,0c1.1,0,2-0.8,2-1.9c0,0,0-0.1,0-0.1C10,6.9,9.2,6,8,6C8.1,6,8,6,8,6z"
              />
            </svg>
            
              appNavigation.configuration
              
            <svg
              aria-hidden="true"
              class="icon-expand"
              fill="currentColor"
              focusable="false"
              height="16"
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8 5L13 10 12.3 10.7 8 6.4 3.7 10.7 3 10z"
              />
            </svg>
          </button>
           
          <transition-stub
            class="nav-item__nav"
            css="true"
            enteractiveclass="collapsing"
            enterclass=""
            entertoclass="collapse show"
            leaveactiveclass="collapsing"
            leaveclass="collapse show"
            leavetoclass="collapse"
          >
            <ul
              class="collapse"
              id="configuration"
              style="display: none;"
            >
              <li
                class="nav-item"
                data-test-id="nav-item-date-time-settings"
              >
                <a
                  class="nav-link"
                  href="/configuration/date-time-settings"
                  target="_self"
                >
                  
                appNavigation.dateTimeSettings
              
                </a>
              </li>
              <li
                class="nav-item"
                data-test-id="nav-item-firmware"
              >
                <a
                  class="nav-link"
                  href="/configuration/firmware"
                  target="_self"
                >
                  
                appNavigation.firmware
              
                </a>
              </li>
              <li
                class="nav-item"
                data-test-id="nav-item-network-settings"
              >
                <a
                  class="nav-link"
                  href="/configuration/network-settings"
                  target="_self"
                >
                  
                appNavigation.networkSettings
              
                </a>
              </li>
              <li
                class="nav-item"
                data-test-id="nav-item-snmp-settings"
              >
                <a
                  class="nav-link"
                  href="/snmp-settings"
                  target="_self"
                >
                  
                appNavigation.snmpSettings
              
                </a>
              </li>
            </ul>
          </transition-stub>
        </li>
        <li
          class="nav-item"
        >
          <button
            aria-controls="access-control"
            aria-expanded="false"
            class="btn btn-link collapsed"
            data-test-id="nav-button-access-control"
            type="button"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              focusable="false"
              height="16"
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 32 32"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M21,2a9,9,0,0,0-9,9,8.87,8.87,0,0,0,.39,2.61L2,24v6H8L18.39,19.61A9,9,0,0,0,30,11.74a8.77,8.77,0,0,0-1.65-6A9,9,0,0,0,21,2Zm0,16a7,7,0,0,1-2-.3l-1.15-.35L17,18.2l-3.18,3.18L12.41,20,11,21.41l1.38,1.38-1.59,1.59L9.41,23,8,24.41l1.38,1.38L7.17,28H4V24.83L13.8,15l.85-.85-.29-.95a7.14,7.14,0,0,1,3.4-8.44,7,7,0,0,1,10.24,6,6.69,6.69,0,0,1-1.09,4A7,7,0,0,1,21,18Z"
              />
              <circle
                cx="22"
                cy="10"
                r="2"
              />
            </svg>
            
              appNavigation.accessControl
              
            <svg
              aria-hidden="true"
              class="icon-expand"
              fill="currentColor"
              focusable="false"
              height="16"
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8 5L13 10 12.3 10.7 8 6.4 3.7 10.7 3 10z"
              />
            </svg>
          </button>
           
          <transition-stub
            class="nav-item__nav"
            css="true"
            enteractiveclass="collapsing"
            enterclass=""
            entertoclass="collapse show"
            leaveactiveclass="collapsing"
            leaveclass="collapse show"
            leavetoclass="collapse"
          >
            <ul
              class="collapse"
              id="access-control"
              style="display: none;"
            >
              <li
                class="nav-item"
                data-test-id="nav-item-ldap"
              >
                <a
                  class="nav-link"
                  href="/access-control/ldap"
                  target="_self"
                >
                  
                appNavigation.ldap
              
                </a>
              </li>
              <li
                class="nav-item"
                data-test-id="nav-item-local-user-management"
              >
                <a
                  class="nav-link"
                  href="/access-control/local-user-management"
                  target="_self"
                >
                  
                appNavigation.localUserManagement
              
                </a>
              </li>
              <li
                class="nav-item"
                data-test-id="nav-item-ssl-certificates"
              >
                <a
                  class="nav-link"
                  href="/access-control/ssl-certificates"
                  target="_self"
                >
                  
                appNavigation.sslCertificates
              
                </a>
              </li>
            </ul>
          </transition-stub>
        </li>
      </ul>
    </nav>
  </div>
   
  <transition-stub
    name="fade"
  >
    <!---->
  </transition-stub>
</div>
`;
